java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -90,19 +92,28 @@
   @VisibleForTesting
   public List<AnalysisError> analysisErrors;
   private int errorsSize = 0;
+
   public SonarComponents(FileLinesContextFactory fileLinesContextFactory, FileSystem fs,
                          JavaClasspath javaClasspath, JavaTestClasspath javaTestClasspath,
                          CheckFactory checkFactory) {
-    this(fileLinesContextFactory, fs, javaClasspath, javaTestClasspath, checkFactory, null);
+    this(fileLinesContextFactory, fs, javaClasspath, javaTestClasspath, checkFactory, null, null);
```

Comment:
This constructor is going to make `workDir()` trigger a NPE... but only used in tests. Should we still mark it as `@Nullable` and null-checked? I don't really know what would be the best. In production we are never going to have it null.

Reply:
As it can be null in sonarlint context i am going to be defensive about this even though that should not be triggered.